src/server/src/main.rs

```diff
@@ -1,23 +1,215 @@
+use bastion::prelude::*;
 use ntex::web;
+use ntex::web::{App, HttpRequest, HttpResponse, HttpServer, Responder};
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use std::sync::{Arc, Mutex};
+use uuid::Uuid;
 
-#[web::get("/")]
-async fn index() -> impl web::Responder {
-    "Hello, World!"
+// user
+#[derive(Deserialize, Serialize, Clone)]
+struct User 
+{
+    username: String,
+    password: String,
+    email: Option<String>,
+    age: Option<u32>,
+    gender: Option<String>,
 }
+// sessionId -> child process reference
+type SessionMap = Arc<Mutex<HashMap<String, ChildRef>>>;
+// username -> User
+type UserMap = Arc<Mutex<HashMap<String, User>>>;
 
-#[web::get("/{name}")]
-async fn hello(name: web::types::Path<String>) -> impl web::Responder {
-    web::HttpResponse::Ok()
-        .content_type("text/plain")
-        //THIS IS FOR THE DEMO, DO NOT DO THIS IN PRODUCTION GET RID OF THIS SOON
-        .set_header(ntex::http::header::ACCESS_CONTROL_ALLOW_ORIGIN, "*")
-        .body(format!("Hello {}!", &name))
+fn spawn_user_session(username: String) -> ChildRef 
+{
+    Bastion::children(|children| 
+        {
+        children.with_exec(move |ctx| 
+            {
+            let username_clone = username.clone();
+            async move 
+            {
+                loop 
+                {
+                    let msg = ctx.recv().await?;
+                    if let Ok(task) = msg.downcast::<String>() 
+                    {
+                        println!("Session for {} doing: {}", username_clone, task);
+                    }
+                    ctx.reply(format!("done {}", username_clone)).unwrap();
+                }
+                Ok(())
+            }
+        })
+    })
+    .expect("Failed to spawn user session")
+}
+
+async fn create_user_handler(
+    body: web::types::Json<User>,
+    path: web::types::Path<(String,)>,
+    user_map: web::types::Data<UserMap>,) -> impl Responder 
+{
+    let new_user = body.into_inner();
+    let (username,) = path.into_inner();
+    let mut map = user_map.lock().unwrap();
+
+    if map.contains_key(&username) {
+        return HttpResponse::build(418).body("User already exists.");
+    }
+
+    map.insert(username.clone(), new_user);
+    HttpResponse::Ok().body("User successfully created.")
```

Comment:
i would turn this last statement into an `if...else...` with both tails using a `return`, just to make it super clear to the reader that if `map.contains_key(&username)` we're done.